config/cache/v1alpha1/cache.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package gingersnap.config.cache.v1alpha1;
+import "apimachinery/pkg/api/resource/quantity.proto";
+
+// TODO  option java_package = define package naming for java
+// TODO  option java_outer_classname = define outer class name
+option java_multiple_files = true;
+// TODO option go_package = verify below value is ok
+option go_package = "api/v1alpha1";
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes the desired configuration for a Cache. Only DB Cache Service is supported atm
+message CacheSpec {
+  // Resource profile for the cache provider
+  ResourcesProfile resources = 1;
+  // DatasourceRef or a ServiceBindingRef (TODO clarify)
+  NamespacedRef dataSourceRef = 2;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  DBType dbType = 1;
+  // Additional properties. DB specific
+  map<string,string> connectionProperties = 2;
+  // Reference to the ServiceBinding CR
+  NamespacedRef serviceBindingRef = 3;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message EagerCachingRuleSpec {
+  // Reference to the related Cache CR
+  NamespacedRef  cacheRef = 1;
+  // Resource profile for the DB sincer
+  ResourcesProfile resources = 2;
+  // Name of the table from where the data will be produced. Format could change depending
+  // on the DB: table or schema.table must be at least supported
+  string tableName = 3;
+  // Format of the key for the get(key) operation
+  KeyFormat keyType = 4;
+  // Separator character in case of plain test key format
+  string keySeparator = 5;
+  // Table columns composing the primary key (where clause)
+  repeated string keyColumns = 6;
+  // Table columns that will be fetched from the DB (select clause)
+  repeated string valueColumns = 7;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message LazyCachingRuleSpec {
+  // Reference to the related Cache CR
+  NamespacedRef cacheRef = 1;
+  // The select query needed to fetch values from the DB
+  string query = 2;
+  // Table columns composing the primary key
+  repeated string keyColumns = 3;
+  // Separator character in case of plain test key format
+  string keySeparator = 4;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a resources profile required for a workload
+message ResourcesProfile {
+  // Memory,CPU resource limit for the cache provider
+  map<string,k8s.io.apimachinery.pkg.api.resource.Quantity> limits = 1;
```

Comment:
What do you envisage the keys to be? I'm curious why we require a map here.

Reply:
yeah, I've imported the Quantity from the [k8s apimachinery](https://github.com/kubernetes/apimachinery/blob/v0.25.3/pkg/api/resource/generated.proto), then I decided to replicate the [k8s design](https://pkg.go.dev/k8s.io/api/core/v1#ResourceList). maybe its a little overkill, on the other hands it's close to the "official" design. No problem in simplifying it anyway.